### PR TITLE
Fix view initialization for more than 2B elements

### PR DESCRIPTION
**Pull request by crtrott**
Fixes issue #2771 

I have no idea why clang-format now decided to change other lines. It was clang 8 also ...

### CODE REVIEW

File: core/src/impl/Kokkos_ViewMapping.hpp [hunk 1]
```diff
@@ -2740,7 +2740,7 @@ struct ViewValueFunctor;
 
 template <class ExecSpace, class ValueType>
 struct ViewValueFunctor<ExecSpace, ValueType, false /* is_scalar */> {
-  typedef Kokkos::RangePolicy<ExecSpace> PolicyType;
+  typedef Kokkos::RangePolicy<ExecSpace, Kokkos::IndexType<int64_t>> PolicyType;
```
Comment (crtrott) [inline core/src/impl/Kokkos_ViewMapping.hpp]: This is one of the two changes which matter

File: core/src/impl/Kokkos_ViewMapping.hpp [hunk 2]
```diff
@@ -2800,7 +2800,7 @@ struct ViewValueFunctor<ExecSpace, ValueType, false /* is_scalar */> {
 
 template <class ExecSpace, class ValueType>
 struct ViewValueFunctor<ExecSpace, ValueType, true /* is_scalar */> {
-  typedef Kokkos::RangePolicy<ExecSpace> PolicyType;
+  typedef Kokkos::RangePolicy<ExecSpace, Kokkos::IndexType<int64_t>> PolicyType;
```
Comment (crtrott) [inline core/src/impl/Kokkos_ViewMapping.hpp]: This is the other one.

File: core/src/impl/Kokkos_ViewMapping.hpp (hunk 1 above)
Comment (dalg24) [reply core/src/impl/Kokkos_ViewMapping.hpp]: `?w=1` at the end of the url in your browser does the trick ;)

**Comment by crtrott** [comment]
No because that is the default and its int for CUDA.

**Comment by dalg24** [comment]
> No because that is the default and its int for CUDA.

I thought the default value the default value was `IndexType<Cuda::size_type>` aka `IndexType<unsigned int>` AFAICT
https://github.com/kokkos/kokkos/blob/3501bae071366609614817c22adf17d824dd942a/core/src/impl/Kokkos_AnalyzePolicy.hpp#L204-L207

What am I missing?

**Comment by crtrott** [comment]
Yeah or that ...

**Comment by crtrott** [comment]
Yeah or that ...

**Comment by crtrott** [comment]
I.e. its unsigned but still its internally overflowing when we hit view lengths of 2.7B

File: core/src/impl/Kokkos_ViewMapping.hpp (hunk 1 above)
Comment (mhoemmen) [reply core/src/impl/Kokkos_ViewMapping.hpp]: @rrdrake @micahahoward @vbrunini It might be worth patching your Trilinos with these changes.

**Pull request merged**